File: backend/src/main/java/com/rowin/contabilidad/services/UsuarioService.java
package com.rowin.contabilidad.services;

import com.rowin.contabilidad.dto.usuario.UsuarioCreateRequest;
import com.rowin.contabilidad.dto.usuario.UsuarioResponse;
import com.rowin.contabilidad.dto.usuario.UsuarioUpdateRequest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface UsuarioService {

    Page<UsuarioResponse> listar(Pageable pageable);

    Page<UsuarioResponse> listarPorEmpresa(Long empresaId, Pageable pageable);

    UsuarioResponse obtenerPorId(Long id);

    UsuarioResponse crear(UsuarioCreateRequest request);

    UsuarioResponse actualizar(Long id, UsuarioUpdateRequest request);

    void eliminar(Long id);

    void cambiarEstado(Long id, boolean active);
}

File: frontend/src/services/usuarioService.ts
import { http } from "./http";

export type UsuarioRolResumen = {
  id: number;
  nombre: string;
};

export type Usuario = {
  id: number;
  createdAt: string;
  updatedAt: string;
  active: boolean;
  empresaId: number;
  username: string;
  email: string;
  roles: UsuarioRolResumen[];
};

export type UsuarioCreateRequest = {
  empresaId: number;
  username: string;
  email: string;
  password: string;
  rolIds?: number[];
};

export type UsuarioUpdateRequest = {
  username: string;
  email: string;
  password?: string;
  rolIds?: number[];
};

export const usuarioService = {
  async listar(page: number = 0, size: number = 20): Promise<{ content: Usuario[]; totalElements: number }> {
    const response = await http.get(`/api/usuarios/listar?page=${page}&size=${size}`);
    return response.data;
  },

  async listarPorEmpresa(empresaId: number, page: number = 0, size: number = 20): Promise<{ content: Usuario[]; totalElements: number }> {
    const response = await http.get(`/api/usuarios/listar-por-empresa/${empresaId}?page=${page}&size=${size}`);
    return response.data;
  },

  async obtenerPorId(id: number): Promise<Usuario> {
    const response = await http.get(`/api/usuarios/${id}`);
    return response.data;
  },

  async crear(data: UsuarioCreateRequest): Promise<Usuario> {
    const response = await http.post("/api/usuarios/crear", data);
    return response.data;
  },

  async actualizar(id: number, data: UsuarioUpdateRequest): Promise<Usuario> {
    const response = await http.put(`/api/usuarios/actualizar/${id}`, data);
    return response.data;
  },

  async eliminar(id: number): Promise<void> {
    await http.delete(`/api/usuarios/eliminar/${id}`);
  },

  async cambiarEstado(id: number, active: boolean): Promise<void> {
    await http.patch(`/api/usuarios/cambiar-estado/${id}?active=${active}`);
  }
};
